aesara/tensor/subtensor.py

```diff
@@ -135,8 +135,8 @@ def as_index_constant(
 
 
 def as_index_literal(
-    idx: Union[Variable, slice, type(np.newaxis)]
-) -> Union[int, slice, type(np.newaxis)]:
+    idx: Optional[Union[Variable, slice]]
+) -> Optional[Union[int, slice]]:
```

Comment:
This doesn't seem necessary.

Reply:
The `type(np.newaxis)` call is not allowed for a type annotation. It gives this error:
```
aesara\tensor\subtensor.py:138: error: Invalid type comment or annotation  [valid-type]
aesara\tensor\subtensor.py:138: note: Suggestion: use type[...] instead of type(...)
```
But with `type[np.newaxis]` there's another error:
```
aesara\tensor\subtensor.py:138: error: Variable "numpy.newaxis" is not valid as a type  [valid-type]
aesara\tensor\subtensor.py:138: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
```
That's probably because `np.newaxis is None`, so `Optional[Union[int, slice]]` remained.

Reply:
What we actually need to do is evaluate `type(np.newaxis)` and use the result in the annotation, so `type[np.newaxis]` shouldn't make sense here.
 
It should be possible to evaluate that expression beforehand (e.g. in a `TYPE_CHECKING` block) and use the result in the annotation, but we can skip that for now and safely assume `np.newaxis` will be `None`.